### PR TITLE
dont fail application is Slack fails to connect

### PR DESCRIPTION
**Pull request by l3rl4n**
We are seeing our entire app go down if there is a Slack Server Issue. 

image

### CODE REVIEW

**Comment by l3rl4n** [comment]
This also addresses the compiler warning
```==> slack
Compiling 10 files (.ex)
warning: variable "error_message" is unused
  lib/slack/sends.ex:101```

File: lib/slack/sends.ex [hunk 1]
```diff
@@ -98,7 +98,8 @@ defmodule Slack.Sends do
       {:ok, response} ->
         case Poison.Parser.parse!(response.body, keys: :atoms) do
           %{ok: true, channel: %{id: id}} -> on_success.(id)
-          e = %{error: error_message} -> on_error.(e)
+          %{error: error_message} -> on_error.(error_message)
```
Comment (BlakeWilliams) [inline lib/slack/sends.ex]: This is changing existing behavior of passing the entire error to just passing the message. I think this line could be:

```elixir
error = %{error: _error_message} -> on_error.(error)
```

File: lib/slack/sends.ex [hunk 2]
```diff
@@ -98,7 +98,8 @@ defmodule Slack.Sends do
...
+          {:error, reason} -> on_error.(reason)
```
Comment (BlakeWilliams) [inline lib/slack/sends.ex]: Can we get a test for this?

**Comment by BlakeWilliams** [comment]
Thanks for submitting this! I added two comments, let me know if you run into any issues there or if you need help figuring out the tests for that scenario. 👍

**Pull request closed without merging**